src/index.ts

```diff
@@ -186,64 +197,38 @@ export class HTTPSnippet {
-          //
-          //  1. The API for `form.append()` has three arguments and the third should only be present when the second is a
-          //    Blob or USVString.
-          //  1. `FormData.pipe()` isn't a function.
-          //
-          // Since the native FormData object is iterable, we easily detect what version of `form-data` we're working with here to allow `multipart/form-data` requests to be compiled under both browser and Node environments.
-          //
-          // This hack is pretty awful but it's the only way we can use this library in the browser as if we code this against just the native FormData object, we can't polyfill that back into Node because Blob and File objects, which something like `formdata-polyfill` requires, don't exist there.
-          // @ts-expect-error TODO
-          const isNativeFormData = typeof form[Symbol.iterator] === 'function';
-
-          // TODO: THIS ABSOLUTELY MUST BE REMOVED.
-          // IT BREAKS SOME USE-CASES FOR MULTIPART FORMS THAT DEPEND ON BEING ABLE TO SET THE BOUNDARY.
-          // easter egg
-          const boundary = '---011000010111000001101001'; // this is binary for "api". yep.
-          if (!isNativeFormData) {
-            // @ts-expect-error THIS IS WRONG.  VERY WRONG.
-            form._boundary = boundary;
-          }
-
           request.postData?.params.forEach(param => {
             const name = param.name;
             const value = param.value || '';
             const filename = param.fileName || null;
+            const contentType = param.contentType || '';
 
-            if (isNativeFormData) {
-              if (isBlob(value)) {
-                // @ts-expect-error TODO
-                form.append(name, value, filename);
-              } else {
-                form.append(name, value);
-              }
+            if (filename) {
+              form.append(name, new Blob([value], { type: contentType }), filename);
             } else {
-              form.append(name, value, {
-                // @ts-expect-error TODO
-                filename,
-                // @ts-expect-error TODO
-                contentType: param.contentType || null,
-              });
+              form.append(name, value);
             }
           });
 
-          if (isNativeFormData) {
-            // eslint-disable-next-line no-restricted-syntax
-            for (const data of formDataIterator(form, boundary)) {
-              request.postData.text += data;
-            }
-          } else {
-            form.pipe(
-              mapStream((data: string) => {
-                request.postData.text += data;
-              }),
-            );
-          }
-
+          const boundary = '---011000010111000001101001'; // this is binary for "api"
           request.postData.boundary = boundary;
+          request.postData.text = await formDataToString(form).then(str => {
```

Comment:
This `formdata-to-string` library I published last week from extracted `undici` code is a goddamn dream. Give it a `FormData` object and it just works without any fuss.

Reply:
Wonder why undici doesnt publish that as a util function.

Reply:
I'm not sure but if they did publish it we'd be importing a whole bunch of junk as their core function has handling for all sorts of body payloads that we don't need to worry about. https://github.com/nodejs/undici/blob/e39a6324c4474c6614cac98b8668e3d036aa6b18/lib/fetch/body.js

Reply:
@kanadgupta I added a custom boundary option in [v1.1.0](https://github.com/readmeio/formdata-to-string/compare/1.0.1...1.1.0).